File: Medium/6. Zigzag Conversion/sol.cs
//  Link: https://leetcode.com/problems/zigzag-conversion/description/

public class Solution {
    // Function to convert the input string into Zigzag pattern based on the given number of rows
    public string Convert(string s, int numRows) {
        // If there's only one row, return the string as is
        if(numRows == 1) return s;

        // Create an array to hold strings for each row
        string [] row = new string[numRows];
        // Initialize each row with an empty string
        for(int i=0; i < numRows; i++){
            row[i]="";
        }

        // Initialize variables to keep track of current row and direction of traversal
        var curRow = 0; // Current row index
        var goingDown = false; // Flag to indicate whether moving downwards or upwards

        // Iterate through each character in the input string
        foreach(char c in s){
            // Append the current character to the string of the current row
            row[curRow] += c;

            // Check if it's time to change direction
            if(curRow == 0 || curRow == numRows - 1){
                goingDown = !goingDown; // Toggle the direction
            }

            // Update the current row index based on the direction
            if(goingDown){
                curRow++; // Move downwards
            }
            else{
                curRow--; // Move upwards
            }
        }

        // Concatenate the strings from each row to form the final result
        string result = "";
        for(int i=0; i < row.Length; i++){
            result += row[i];
        }
        // Return the Zigzag pattern as a single string
        return result;
    }
}

File: Medium/260. Single Number III/sol.ts
// Link: https://leetcode.com/problems/single-number-iii/description/

function singleNumber(nums: number[]): number[] {
    let xy = 0;
    // XOR all numbers to get the XOR of the two unique numbers
    for (let n of nums) {
        xy ^= n;
    }

    // Get the rightmost set bit of xy
    xy &= -xy;

    // Initialize the result array to store the two unique numbers
    const result = [0, 0];

    // Divide the numbers into two groups and XOR within each group
    for (let n of nums) {
        if ((xy & n) === 0) {
            result[0] ^= n;
        } else {
            result[1] ^= n;
        }
    }

    return result;
}
